### PR TITLE
Finish rest exercise

### PR DESCRIPTION
**Pull request by shaharshw**

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+@Controller
+@Path("/api/owner")
+@Produces(MediaType.APPLICATION_JSON)
+class OwnerResource {
+
+    @GET
+    fun getALLOwners() : Response {
+        return Response.ok(listOf(
+            Owner(1, "Shahar", 1, 1, null,null),
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt]:
```suggestion
            Owner(1, "Shahar", 1, 1, null, null),
```

File: src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            Owner(2, "Alex", 1, 2, null,null )
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt]:
```suggestion
            Owner(2, "Alex", 1, 2, null, null)
```

File: src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            .build()
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun createOwner(owner: Owner): Response {
+
+        owner.name = owner.name ?: run {
+            if (owner.firstName.isNullOrEmpty() || owner.lastName.isNullOrEmpty()) {
+                return Response.status(Status.BAD_REQUEST).entity("Name is required").build()
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt]: Nice! You can also just throw a BadInputException("Name is required"). The framework catches the BadInputException and handles the Response.status for you.

File: src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            } else {
+                "${owner.firstName} ${owner.lastName}"
+            }
+        }
+
+        owner.name?.let { fullName ->
+            val nameParts = fullName.split(" ")
+            owner.firstName = nameParts[0]
+            owner.lastName = nameParts.getOrNull(1)
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt]: Something to consider. What happens if they don't send a last name. Also what happens if he has multiple spaces? So Jackie Van Halen for example, you would miss out on the rest of his name.

Comment (shaharshw) [reply src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt]: If they don't provide a last name, I need to reject it, right? As for the second issue, I fixed it.

**Pull request merged**